### PR TITLE
Clarify homeserver configuration and requirements in documentation

### LINKED ISSUE #238: Improve configurations documentation for setup of homeserver
**Is your feature request related to a problem? Please describe.**
For startes it's not clear how in detail to configure the service. E.g. 

```
bridge:
  # Basic homeserver configuration
  #
  domain: example.com
  url: http://localhost:8008
  mediaUrl: http://example.com
  port: 9993
  bindAddress: 127.0.0.1
```

does this relate to an existing homeserver or is this a homeserver the hookshot is setting up

**Describe the solution you'd like**

Be more verbose and maybe even state some of the obvious things

### PR DESCRIPTION
**Pull request by Half-Shot**
Fixes #238

### CODE REVIEW

File: docs/setup.md [hunk 1]
```diff
@@ -60,6 +63,23 @@ Copy `registration.sample.yml` into `registration.yml` and fill in:
...
+### Homeserver Configuration
+
+In addition to providing the registration file above, you also need to tell Hookshot how to reach your homeserver. For clarity, hookshot
+expects to be able to connect to an existing homeserver which has it's registration file configured.
+
+```yaml
+bridge:
+  domain: example.com # your homeserver's server name
+  url: http://localhost:8008 # The URL where hookshot can reach the client-server API.
+  mediaUrl: http://example.com # Optional. The url where media hosted on your homeserver is reachable (this should be publically reachable from the internet)
```
Comment (HarHarLinks) [inline docs/setup.md:75]:
```suggestion
  mediaUrl: https://example.com # Optional. The url where media hosted on your homeserver is reachable (this should be publically reachable from the internet)
```

**Review by HarHarLinks — CHANGES_REQUESTED** [review]

File: docs/setup.md [hunk 2]
```diff
@@ -60,6 +63,23 @@ Copy `registration.sample.yml` into `registration.yml` and fill in:
...
 You will need to link the registration file to the homeserver. Consult your homeserver documentation
 on how to add appservices. [Synapse documents the process here](https://matrix-org.github.io/synapse/latest/application_services.html).
+### Homeserver Configuration
+
+In addition to providing the registration file above, you also need to tell Hookshot how to reach your homeserver. For clarity, hookshot
+expects to be able to connect to an existing homeserver which has it's registration file configured.
+
+```yaml
+bridge:
+  domain: example.com # your homeserver's server name
```
[the remark this answers is not in the document]
Comment (Half-Shot) [reply docs/setup.md:73]: Yep, it's actually official terminology as per https://spec.matrix.org/v1.2/appendices/#server-name

[the remark this answers is not in the document]
Comment (HarHarLinks) [reply docs/setup.md:73]:
```suggestion
  domain: example.com # your homeserver's server name per https://spec.matrix.org/v1.2/appendices/#server-name
```

[the remark this answers is not in the document]
Comment (frlan) [reply docs/setup.md:73]: Just to be clear -- "your" is a very wide area. I'd rather see something like "The homeserver hookshot shall communicate with (or similar)

**Comment by frlan** [comment]
Yes, better.

File: docs/setup.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (Half-Shot) [reply docs/setup.md:73]: I would rather the latter, I worry linking the spec at each technical term is going to excessively bog down and confuse users even more. This documentation is also definitely targetted towards people who have:

A) Setup a homeserver themselves
B) Have a "good" understanding of Matrix (not a developer, but enough to understand the concept of multiple servers and the wider federation).

I'll accept we could clarify the "your" here to something more specific.

[the remark this answers is not in the document]
Comment (Half-Shot) [reply docs/setup.md]: Have updated the sentence above the snippet to clarify which homeserver we are talking about. I think making anymore changes here feels like hypotethical cases of confusion and I would like to see how this turns out in practice :)

**Pull request merged**